pkg/api/http_client.go

```diff
@@ -135,7 +135,14 @@ func isGarage(host string) bool {
 }
 
 func isEnterprise(host string) bool {
-	return host != github && host != localhost
+	return host != github && host != localhost && !isTenancy(host)
+}
+
+// tenancyHost is the domain name of a tenancy GitHub instance.
+const tenancyHost = "ghe.com"
+
+func isTenancy(host string) bool {
+	return strings.HasSuffix(host, "."+tenancyHost)
```

Comment:
**Are there any concerns about this being seemingly redundant to the logic in `auth`?**

I did a double take when I first saw this because I thought the enterprise check was already taking tenancy into account, when I realized I was thinking about:

https://github.com/cli/go-gh/blob/25db6b99518c88e03f71dbe9e58397c4cfb62caf/pkg/auth/auth.go#L152-L167

My concern is managing 2 sets of similar / overlapping functionality in 2 places and introducing drift.

Reply:
See "implementation choices" section in PR description.

I think we should do work here or in a follow up to address this but I'm not going through tearing things up until we know this is works as expected and covers the right use cases.